### ISSUE TITLE
Single Item last field gets duplicated on Replace (When upsert: true and field names in Datastore are in camel case)

### ISSUE THREAD

**Issue by russkyc** [ttu/json-flatfile-datastore#87]
## Bug:
- When replacing a field value on single item mode, the last field that gets updated duplicates itself in the DataStore file.

## Replication:
```csharp
public static class Program
{
    public static void Main(string[] args)
    {
        var entity = new Entity();
        
        for (int i = 1; i <= 2; i++)
        {
            // First to Update
            entity.DoesNotDuplicate = $"Updated {i} Times, Doesn't Duplicate";
                    
            // Last to Update
            entity.Duplicates = $"Updated {i} Times, Duplicates";
            Thread.Sleep(500);
        }
    }
    internal class Entity
    {
        private static readonly DataStore _dataStore = new DataStore("DuplicateBug.json");

        internal string? DoesNotDuplicate
        {
            get => _dataStore.GetItem(nameof(DoesNotDuplicate));
            set => _dataStore.ReplaceItem(nameof(DoesNotDuplicate), value, true);
        }

        internal string? Duplicates
        {
            get => _dataStore.GetItem(nameof(Duplicates));
            set => _dataStore.ReplaceItem(nameof(Duplicates), value, true);
        }
    }
}
```

## Output:
```json
{
  "doesNotDuplicate": "Updated 2 Times, Doesn't Duplicate",
  "duplicates": "Updated 1 Times, Duplicates",
  "duplicates": "Updated 2 Times, Duplicates"
}
```

## Observation:
- the doesNotDuplicate field is updated correctly, does not duplicate itself since it is the first field updated.
- the duplicates field is not updated correctly, and duplicates itself instead.

**Comment by ttu** [comment]
Thanks for the issue!

I was able to reproduce the problem and it is due to DataStore’s internal handling of key strings. Keys are converted to selected format (e.g. lower camel case) just before saving the JSON, so it is possible that internal data structure can have collection/item-keys starting with upper and lowercase letter, which will then lead to duplicate items on the next save.

I will make a fix for this and release a patch version of the package.

**Comment by russkyc** [comment]
Thanks @ttu ! This is awesome!

**Comment by ttu** [comment]
Thanks for the good example and observations. With those reproducing the error was easy 👍 

Failing test cases: https://github.com/ttu/json-flatfile-datastore/commit/d6410a7dc03fa23ac4997e549c4e0819ccd8a8a9

Fix coming next 😄

**Comment by ttu** [comment]
Fixed in: #89 

Fix will be in the next patch release.

**Comment by ttu** [comment]
Fix released: https://www.nuget.org/packages/JsonFlatFileDataStore/2.4.2
